### DISCUSSION TITLE
Using semantic HTML roles in addition to elements

### DISCUSSION THREAD

**Discussion by ouvreboite** [beercss/beercss#377] [Ideas]
Hi 👋,  first thank you for beerCSS, that’s an awesome library.

I’m personally more of a backend dev, but I stumbled over various « semantic » CSS libraries (beerCSS, picoCSS…) which have renewed my interest in frontend.
Most notably because the focus on HTML semantic give me the impression that using those libraries teach me HTML best-practices instead of teaching me 100+ classes to put on `<div>` :)

In the same spirit, what do you think about using the available HTML roles and aria attributes (when they exists) instead of library specific classes?

For example, beerCSS’ [tabs](https://github.com/beercss/beercss/blob/main/docs/TABS.md) are currently structured as such:

```html
<div class="tabs">
  <a data-ui="#page1" class="active">Tab 1</a>
  <a data-ui="#page2">Tab 2</a>
  <a data-ui="#page3">Tab 3</a>
</div>

<div class="page" id="page1">
  <h5>Tab 1</h5>
</div>
<div class="page" id="page2">
  <h5>Tab 2</h5>
</div>
<div class="page" id="page3">
  <h5>Tab 3</h5>
</div>
```

It relies only on beerCSS specific classes and attributes.

But there are already official roles for [tabs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tablist_role): tablist, tab and tabpanel, and some related aria attributes.

Using those, the same tabs structure could be described as this:

```html
<div role="tablist">
  <a role="tab" aria-controls="page1" aria-selected="true">Tab 1</a>
  <a role="tab" aria-controls="page2">Tab 2</a>
  <a role="tab" aria-controls="page3">Tab 3</a>
</div>

<div role="tabpanel" id="page1">
  <h5>Tab 1</h5>
</div>
<div role="tabpanel" id="page2">
  <h5>Tab 2</h5>
</div>
<div role="tabpanel" id="page3">
  <h5>Tab 3</h5>
</div>
```

**Comment by acroyear** [comment]
yeah, i kinda like this idea, and it could probably be done 'additive' to the current since it just is a different selector producing the same desired effect. It would be nice if CSS had a role selector short-cut like they do . and # for class and id, but oh well.
